### PR TITLE
Prevent navigation above the root of PhysicalFileProvider

### PR DESCRIPTION
**Pull request by cesarbs**
cc @blowdart

### CODE REVIEW

File: src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs [hunk 1]
```diff
@@ -57,14 +57,49 @@ public void Dispose()
...
             {
                 return null;
             }
+
             return fullPath;
         }
 
+        private bool PathNavigatesAboveRoot(string path)
+        {
+            var tokenizer = new StringTokenizer(path, new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar });
```
Comment (pranavkm) [inline src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs]: Cache the array?

File: src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs [hunk 2]
```diff
@@ -57,14 +57,49 @@ public void Dispose()
 
         private string GetFullPath(string path)
         {
+            if (PathNavigatesAboveRoot(path))
```
Comment (pranavkm) [inline src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs]: Wouldn't `IsUnderneathRoot` take care of rooting this? If not, could you add a comment about scenarios this method prevents that `IsUnderneathRoot` does not?

Comment (Eilon) [inline src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs]: Talk to @blowdart directly. This is needed.

File: src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs (hunk 1 above)
Comment (cesarbs) [inline src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs]: :+1:

**Pull request merged**